src/tests/python/sssd/testlib/common/libdirsrv.py

```diff
@@ -189,31 +194,38 @@ def setup_certs(self, ssl_dir):
         try:
             self.multihost.run_command(stop_ds)
         except subprocess.CalledProcessError:
-            raise DirSrvException("Unable to stop Directory Server instance")
+            raise DirSrvException("Failed to stop Directory Server instance")
         else:
             self.multihost.log.info('DS instance stopped successfully')
             self._copy_pkcs12(ssl_dir)
         cacert_file_path = '%s/cacert.pem' % ('/etc/openldap/cacerts')
         target_pin_file = os.path.join(self.dsinst_path, 'pin.txt')
         pwfile = os.path.join(self.dsinst_path, 'pwfile')
         ca_p12 = os.path.join(self.dsinst_path, 'ca.p12')
-        server_p12 = os.path.join(self.dsinst_path, 'server.p12')
-        # recreate the database
-        certutil_cmd = 'certutil -N -d %s -f %s' % (self.dsinst_path, pwfile)
+        server_p12_name = '%s-%s' % (self.multihost.sys_hostname, 'server.p12')
+        server_p12 = os.path.join(self.dsinst_path, server_p12_name)
+        # create directory to copy ca cert
+        certutil_cmd = 'certutil -T -d %s -f %s' % (self.dsinst_path, pwfile)
         self.multihost.run_command(certutil_cmd)
         create_cert_dir = 'mkdir -p /etc/openldap/cacerts'
         # recreate the database
         self.multihost.run_command(create_cert_dir)
+        if not canick:
+            canick = "ExampleCA"
         pkcs12_file = [ca_p12, server_p12]
         for pkcs_file in pkcs12_file:
             if not self._import_certs(pkcs_file, pwfile):
                 raise DirSrvException("importing certificates failed")
-        set_trust_cmd = 'certutil -M -d %s -n "ExampleCA"'\
-                        ' -t "CTu,u,u" -f %s' % (self.dsinst_path, pwfile)
+        set_trust_cmd = 'certutil -M -d %s -n %s -t ' \
+                        '"CTu,u,u" -f %s' % (self.dsinst_path, canick, pwfile)
```

Comment:
Hi,

just a minor nitpick, I think it would be better readable to not split the option `-t` and to option value `"CTu,u,u".

Additionally I wonder a bit about the `u` in the trust arguments. `T`stands for "trusted CA to issue client certs (implies c)", so it indicates a CA certificate, while `u`indicates a user certificate. Since this is already in the old code and will do no harm it is ok to keep it as is for the time being. But if time permits it might be worth to check where this is coming from and if `u` is really needed here.

bye,
Sumit